packages/app/src/cli/models/app/app.ts

```diff
@@ -97,6 +98,21 @@ export function getAppScopes(config: AppConfiguration) {
   }
 }
 
+export function usesLegacyScopesBehavior(app: AppInterface | AppConfiguration) {
```

Comment:
if a user hasn't opted-in to config in code, or if they have the legacy scopes flag, we use legacy behavior